keystoremanager/keystoremanager.go

```diff
@@ -34,14 +34,29 @@ const (
 	rsaRootKeySize    = 4096 // Used for new root keys
 )
 
-var (
-	// ErrValidationFail is returned when there is no trusted certificate in any of the
-	// root keys available in the roots.json
-	ErrValidationFail = errors.New("could not validate the path to a trusted root")
-	// ErrRootRotationFail is returned when we fail to do a full root key rotation
-	// by either failing to add the new root certificate, or delete the old ones
-	ErrRootRotationFail = errors.New("could not rotate trust to a new trusted root")
-)
```

Comment:
I think having a custom error type here is good, but maybe keep predefined errors for the different reasons so a caller can distinguish between the reasons for the failure without resorting to string comparisons. This would also make the tests cleaner.